### PR TITLE
Add decorator

### PR DESCRIPTION
**Pull request by mwafrika**
## Description

In this project, we worked on creating a database based on the diagram.

### General requirements
- [X] Make sure that there are [no linter errors](https://github.com/microverseinc/linters-config)
- [x] Followed the **Github flow**.
- [x] The documentation is professional.

### Project requirements

- [x] Check the usage of [Decorator pattern in Ruby](https://refactoring.guru/design-patterns/decorator/ruby/example#example-0)
- [x] Think about how you can use two decorators in order to capitalize and trim people's names.
- [x] In the following steps you will get instructions for implementing the Decorator design pattern for this case.

### Thank you for taking the time to review this pull request.

### CODE REVIEW

File: capitalize_decorator.rb [hunk 1]
```diff
@@ -0,0 +1,11 @@
+require_relative 'decorator'
+class CapitalizeDecorator < Decorator
+  def initialize(nameable)
+    super()
+    @nameable = nameable
+  end
+
+  def correct_name
+    @nameable.correct_name
+  end
```
Comment (ShoiraTa) [inline capitalize_decorator.rb]: - [ ] `correct_name` function should capitalize the first letter of the given name. Therefore should looks like this: 
```ruby
    def correct_name
      @nameable.correct_name.capitalize
    end
```

File: trimmer_decorator.rb [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+    super()
+    @nameable = nameable
+  end
+
+  def correct_name
+    if @nameable.correct_name.length > 10
+      @nameable.correct_name[0...10]
+    else
+      @nameable.correct_name
+    end
```
Comment (ShoiraTa) [inline trimmer_decorator.rb]: - [ ] OPTIONAL: you can simply write the logic in one line using ternary operator:
```ruby
    def correct_name
      @nameable.correct_name.length > 10 ? @nameable.correct_name[0..9] : @nameable.correct_name
    end
```

**Review by ShoiraTa — CHANGES_REQUESTED** [review]
Hi @mwafrika,

:thumbsup: This is good to make it this far, your project is really nice, congratulations on the good work you have done so far.

**First let me to point out:**

- You passed all the Linters! ✔️ 
- Professional pull request! ✔️ 

Keep on coding, good job so far! :clap:

There are some issues that you still need to work on to go to the next project but you are almost there!



 ## Required Changes  :recycle: 

_Check the comments under the review._ 

***
Cheers and Happy coding! :clap: :clap: :clap:

Feel free to leave any questions or comments in the PR thread if something is not 100% clear by mentioning my Github username.


**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

* * *
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by DeliceLydia — APPROVED** [review]
# Hi @mwafrika, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
# Congratulations! 🎉 



### Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**